src/Microsoft.Framework.Runtime.Loader/LoadContext.cs

```diff
@@ -224,10 +227,30 @@ private Assembly ResolveAssembly(object sender, ResolveEventArgs args)
                     return null;
                 }
 
-                if (args.RequestingAssembly != null)
+                var requestingAssembly = args.RequestingAssembly;
+
+#if ASPNET50
+                // On Mono while loading the !preprocess assembly the ResolveEventArgs.RequestingAssembly is not populated. 
+                // As a result when code in !preprocess references a class library the load fails on the class library.
+                // Below is a work around to try populate the requesting assembly.
+
+                if (Microsoft.Framework.Runtime.PlatformHelper.IsMono &&
+                    requestingAssembly == null &&
+                    !assemblyName.Name.Contains("!"))
+                {
+                    // See if this loading is for !preprocess.
+                    var domain = (AppDomain)sender;
+
+                    requestingAssembly = domain.GetAssemblies()
```

Comment:
I couldn't find a better place to hack this and this looked like the simplest fix with minimal impact. This code automatically no-ops if mono fixes the bug as `requestingAssembly` becomes non `null`.